Mage.Sets/src/mage/cards/d/DemonOfFatesDesign.java

```diff
@@ -0,0 +1,261 @@
+        super(card);
+    }
+
+    @Override
+    public DemonOfFatesDesign copy() {
+        return new DemonOfFatesDesign(this);
+    }
+}
+
+class DemonOfFatesDesignPayLifeCost extends PayLifeCost {
+
+    private MageObjectReference mor;
+
+    DemonOfFatesDesignPayLifeCost(int amount) {
+        super(amount);
+    }
+
+    private DemonOfFatesDesignPayLifeCost(final DemonOfFatesDesignPayLifeCost cost) {
+        super(cost);
+        this.mor = cost.mor;
+    }
+
+    @Override
+    public DemonOfFatesDesignPayLifeCost copy() {
+        return new DemonOfFatesDesignPayLifeCost(this);
+    }
+
+    MageObjectReference getMor() {
+        return this.mor;
+    }
+
+    void setMor(MageObjectReference mor) {
+        this.mor = mor;
+    }
+
+}
+
+enum DemonOfFatesDesignCost implements DynamicCost {
+    instance;
+
+    @Override
+    public Cost getCost(Ability ability, Game game) {
+        return new DemonOfFatesDesignPayLifeCost(ability.getManaCosts().manaValue());
+    }
+
+    @Override
+    public String getText(Ability ability, Game game) {
+        return "Pay " + ability.getManaCosts().manaValue() + " life rather than "
+                + ability.getManaCosts().getText() + '?';
+    }
+}
+
+class DemonOfFatesDesignAlternativeCostSourceAbility extends AlternativeCostSourceAbility {
+
+    private MageObjectReference mor;
+
+    DemonOfFatesDesignAlternativeCostSourceAbility() {
+        super(
+                new CompoundCondition(SourceIsSpellCondition.instance, IsBeingCastFromHandCondition.instance),
+                null, StaticFilters.FILTER_CARD_ENCHANTMENT,
+                true, DemonOfFatesDesignCost.instance
+        );
+    }
+
+    private DemonOfFatesDesignAlternativeCostSourceAbility(final DemonOfFatesDesignAlternativeCostSourceAbility effect) {
+        super(effect);
+    }
+
+    @Override
+    public DemonOfFatesDesignAlternativeCostSourceAbility copy() {
+        return new DemonOfFatesDesignAlternativeCostSourceAbility(this);
+    }
+
+    @Override
+    protected void doActivate(Game game, Ability ability) {
+        super.doActivate(game, ability);
+        for (Cost cost : ability.getCosts()) {
+            if (cost != null && cost instanceof DemonOfFatesDesignPayLifeCost) {
+                ((DemonOfFatesDesignPayLifeCost) cost).setMor(getMor(game));
+            }
+        }
+    }
+
+    MageObjectReference getMor(Game game) {
+        return new MageObjectReference(getSourceObject(game), game);
+    }
+}
+
+class DemonOfFatesDesignCastEffect extends ContinuousEffectImpl {
+
+    private final DemonOfFatesDesignAlternativeCostSourceAbility alternativeCastingCostAbility
+            = new DemonOfFatesDesignAlternativeCostSourceAbility();
+
+    DemonOfFatesDesignCastEffect() {
+        super(Duration.WhileOnBattlefield, Outcome.Neutral);
+        this.staticText = "You may cast an enchantment spell by paying life equal to its mana value rather than paying its mana cost.";
```

Comment:
don't capitalize first letter of staticText here, should just be "you may cast"...